itest/mint_batch_stress_test.go

```diff
@@ -30,29 +31,109 @@ var (
 )
 
 func testMintBatch100StressTest(t *harnessTest) {
-	mintBatchStressTest(t, 100, defaultWaitTimeout)
+	timeout := defaultWaitTimeout
+
+	// If we create a second tapd instance and sync the universe state,
+	// the synced tree should match the source tree.
+	bob := setupTapdHarness(
+		t.t, t, t.lndHarness.Bob, nil,
+	)
+	defer func() {
+		require.NoError(t.t, bob.stop(!*noDelete))
+	}()
+
+	mintBatches := func(reqs []*mintrpc.MintAssetRequest) []*taprpc.Asset {
+		return mintAssetsConfirmBatch(
+			t, t.tapd, reqs, withMintingTimeout(timeout),
+		)
+	}
+
+	getImageMetadataBytes := func() []byte {
+		return GetImageMetadataBytes(t.t, ImageMetadataFileName)
+	}
+
+	mintBatchStressTest(
+		t.t, t.tapd, bob, t.tapd.rpcHost(), 100, mintBatches,
+		getImageMetadataBytes, timeout,
+	)
 }
 
 func testMintBatch1kStressTest(t *harnessTest) {
-	mintBatchStressTest(t, 1000, defaultWaitTimeout*10)
+	timeout := defaultWaitTimeout * 10
+
+	// If we create a second tapd instance and sync the universe state,
+	// the synced tree should match the source tree.
+	bob := setupTapdHarness(
+		t.t, t, t.lndHarness.Bob, nil,
+	)
+	defer func() {
+		require.NoError(t.t, bob.stop(!*noDelete))
+	}()
+
+	mintBatches := func(reqs []*mintrpc.MintAssetRequest) []*taprpc.Asset {
+		return mintAssetsConfirmBatch(
+			t, t.tapd, reqs, withMintingTimeout(timeout),
+		)
+	}
+
+	getImageMetadataBytes := func() []byte {
+		return GetImageMetadataBytes(t.t, ImageMetadataFileName)
```

Comment:
Will we need to inject some special metadata files later on? Or could we just read the bytes here and then pass them into `mintBatchStressTest` directly?

Reply:
I agree the pattern looks weird now with the simple test cases, but could be useful for injecting different metadata later on for a larger matrix of batch sizes + metadata sizes or types, or randomized metadata.